stdlib/public/core/CMakeLists.txt

```diff
@@ -247,35 +245,40 @@ else()
   # effort has been completed.
   #set(LINK_FLAGS
   #  -Wl,--whole-archive swiftRuntime -Wl,--no-whole-archive)
-  if("${SWIFT_PATH_TO_LIBICU_BUILD}" STREQUAL "")
-    list(APPEND swift_core_private_link_libraries
-      ${SWIFT_${SWIFT_PRIMARY_VARIANT_SDK}_${SWIFT_PRIMARY_VARIANT_ARCH}_ICU_UC}
-      ${SWIFT_${SWIFT_PRIMARY_VARIANT_SDK}_${SWIFT_PRIMARY_VARIANT_ARCH}_ICU_I18N})
+endif()
+
+function(swift_core_private_libraries sdk arch libraries_var_name)
```

Comment:
I think that this is the going in the exact opposite direction of where we want to go.  We need to start using `find_library` and imported targets to create the dependencies rather than hard coded names and magic flags.

Reply:
It is not clear why you placed this review comment here: this reorganizes a chunk of the following CMake config as a CMake function and is orthogonal to the issue of "imported targets" vs "magic flags." Maybe you meant to place it below, where the flags are?

As for the way those libraries and flags are added below, that predates this pull and has nothing to do with me. I realize you'd like to see it modernized, but that's not what this pull is offering: I simply refactor the way CMake calls the pre-existing "magic flags," so they can be used when cross-compiling too.